include/rmm/mr/device/detail/stream_ordered_memory_resource.hpp

```diff
@@ -505,10 +484,6 @@ class stream_ordered_memory_resource : public crtp<PoolResource>, public device_
   // bidirectional mapping between non-default streams and events
   std::unordered_map<cudaStream_t, stream_event_pair> stream_events_;
 
-  // shared pointers to events keeps the events alive as long as either the thread that created
-  // them or the MR that is using them exists.
-  std::set<std::shared_ptr<event_wrapper>> default_stream_events;
```

Comment:
I don't think we can get rid of this.  This is not the same as the TLS events.

Reply:
The only reference in the source was in `get_event` function in the PTDS case, so it didn't seem to be necessary any more.

Reply:
Yes. It ensured that if the thread that owns the TLS exits, but the memory resource still exists, then the default stream events won't be deleted (by the TLS). We could have a memory resource shared by multiple threads using the same device.

Reply:
But now I see that now `cudaEventCreateWithFlags` is called but `cudaEventDestroy` is never called. This "keep alive" set was only needed because of the RAII `event_wrapper`. Eliminating this is a nice side effect I hadn't thought of. :)

Reply:
I added a suggested comment at the creation event to indicate this.